src/Toolbar.tsx

```diff
@@ -1,19 +1,451 @@
-import React, { useState } from 'react';
+/* eslint-disable no-multi-assign */
+/* eslint-disable prefer-destructuring */
+import React, { useRef, useState, useEffect } from 'react';
+import styled from '@emotion/styled';
+import { css } from '@emotion/react';
+import { BiBold, BiUnderline, BiItalic, BiLink } from 'react-icons/bi';
 
 interface ToolbarProps {
   containerRef: React.MutableRefObject<HTMLDivElement | null>;
 }
 
-const Toolbar = ({ containerRef }: ToolbarProps) => {
-  const openTitleMenu = () => {};
+function Toolbar({ containerRef }: ToolbarProps) {
+  const [toolbarPosition, setToolbarPosition] = useState<[number, number]>([0, 0]);
+  const [showToolbar, setShowToolbar] = useState<boolean>(false);
+  const toolbarRef = useRef<HTMLDivElement | null>(null);
 
-  const setBold = () => {};
+  useEffect(() => {
+    document.addEventListener('pointerup', (e) => {
+      const selection: Selection | null = window.getSelection();
 
-  const setUnderLine = () => {};
+      if (selection && selection.type === 'Caret') {
+        setShowToolbar(false);
+      }
 
-  const setItalic = () => {};
+      if (selection && selection.type === 'Range') {
+        const range = selection.getRangeAt(0);
+        if (containerRef.current && containerRef.current.contains(range.commonAncestorContainer)) {
+          calculateToolBarPosition();
+        } else {
+          setShowToolbar(false);
+        }
+      } else {
+        setShowToolbar(false);
+      }
+    });
+  }, [containerRef]);
```

Comment:
```suggestion
  useEffect(() => {
    const cb = () => {
      const selection = window.getSelection();

      console.log(selection?.toString());
    };

    document.addEventListener("selectionchange", cb);

    return () => {
      document.removeEventListener("selectionchange", cb);
    };
  }, []);
```

useEffect 에서 addEventListener 뒤에 clean up 함수에서 remove 하면 해당 컴포넌트가 화면에서 없어질때 불필요한 이벤트도 삭제해주겠네요~ 

그리고 deps 안에는 [] 빈배열을 넣어 마운트, 언마운트 시에만 작동하게 만들면 좋을것 같아요.